Sources/AutobahnDescription/AutobahnDescription.swift

```diff
@@ -60,37 +62,41 @@ public final class Autobahn<H: Highway> {
     
     // MARK: - Public Functions
     
-    public init(_ highwayType: H.Type) {}
+    public init(_ highwayType: H.Type) {
+        self.highwayStart = Date()
+    }
     
     public func beforeAll(handler: @escaping (H) throws -> Void) -> Autobahn<H> {
-        guard self.beforeAllHandler == nil else {
-            fatalError("beforeAll declared more than once")
-        }
-        self.beforeAllHandler = handler
+        precondition(beforeAllHandler == nil, "beforeAll declared more than once")
+        beforeAllHandler = handler
         return self
     }
     
-    public func highway(_ highway: H, dependsOn: [H]? = nil, taskHandler: @escaping () throws -> Void) -> Autobahn<H> {
-        self.highways[highway] = taskHandler
-        if let d = dependsOn {
-            self.dependings[highway] = d
-        }
+    public func highway(_ highway: H, handler: @escaping () throws -> Void) -> Autobahn<H> {
+        self.highways[highway] = handler
+        return self
+    }
+
+    public func highway(_ highway: H, dependsOn highways: [H]) -> Autobahn<H> {
+        dependings[highway] = highways
+        return self
+    }
+
+    public func highway(_ highway: H, dependsOn highways: [H], handler: @escaping () throws -> Void) -> Autobahn<H> {
```

Comment:
Since the `set` subscript takes optionals we should still define this as one function with dependsOn and handler being optional with a default `nil` implementation. Only pitfall: it would be possible to define this:

```Swift
.highway(.build)
```

which would be useless.

Reply:
Is this a compromise we want to make?  One thing I could do is define:
```swift
private func highway(_ highway: H, dependsOn highways: [H]? = nil, handler: (() throws -> Void)? = nil) -> Autobahn<H>
```
as private method and have the other variations forward to this method, that way we only need to update logic in one place

Reply:
That's a good idea 👍